File: src/MisteryApp.Abstractions/Interfaces/IReportService.cs
using MisteryApp.Abstractions.Models;

namespace MisteryApp.Abstractions.Interfaces;

public interface IReportService
{
    Task<WeeklyReport> GetWeeklyReportAsync(int userId, DateOnly weekStart, CancellationToken cancellationToken);
    Task<MonthlyReport> GetMonthlyReportAsync(int userId, DateOnly monthStart, CancellationToken cancellationToken);
}

File: src/misteryapp.web.client/src/domain/interfaces/i-report-service.ts
import type { WeeklyReport, MonthlyReport } from '../models'

export interface IReportService {
  getWeeklyReport(userId: number, weekStart: string, signal?: AbortSignal): Promise<WeeklyReport | null>
  getMonthlyReport(userId: number, monthStart: string, signal?: AbortSignal): Promise<MonthlyReport | null>
}
